### PR TITLE
ulp:feature added espidf-ulp-adc example as minimal build

### PR DESCRIPTION
**Pull request by jhonnyam**
copied the example code and adapted it

### CODE REVIEW

File: platformio.ini [hunk 1]
```diff
@@ -11,10 +11,7 @@
 platform = espressif32
 framework = espidf
 
-[env:pico32]
-board = pico32
-monitor_speed = 115200
-
 [env:nodemcu]
```
Comment (nopeslide) [inline platformio.ini:15]: should be `debug` or `release`

File: platformio.ini [hunk 2]
```diff
@@ -11,10 +11,7 @@
...
 board = nodemcu-32s
```
Comment (nopeslide) [inline platformio.ini:16]: move this into globale env and set it to the actual board

File: platformio.ini [hunk 3]
```diff
@@ -11,10 +11,7 @@
...
-monitor_speed = 115200
\ No newline at end of file
+monitor_speed = 115200
+build_flags = -DCONFIG_WIFI_SSID=\"ESP_AP\" -DCONFIG_WIFI_PASSWORD=\"MYPASS\"
```
Comment (nopeslide) [inline platformio.ini]:
```suggestion
build_flags =
  -DCONFIG_WIFI_SSID=\"ESP_AP\"
  -DCONFIG_WIFI_PASSWORD=\"MYPASS\"
```
it's more readable this way. but please check if an empty first line is permitted

File: src/ulp_adc_example_main.c [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        printf("Thresholds:  low=%d  high=%d\n", ulp_low_thr, ulp_high_thr);
+        ulp_last_result &= UINT16_MAX;
+        printf("Value=%d was %s threshold\n", ulp_last_result,
+                ulp_last_result < ulp_low_thr ? "below" : "above");
+    }
+    printf("Entering deep sleep\n\n");
+    start_ulp_program();
+    ESP_ERROR_CHECK( esp_sleep_enable_ulp_wakeup() );
+    esp_deep_sleep_start();
+}
```
Comment (nopeslide) [inline src/ulp_adc_example_main.c]: put `app_main` in an extra file int the src root dir

**Review by nopeslide — CHANGES_REQUESTED** [review]

File: platformio.ini [hunk 5]
```diff
@@ -14,6 +14,12 @@ framework = espidf
...
 monitor_speed = 115200
+<<<<<<< HEAD
+build_flags = 
+    -DCONFIG_WIFI_SSID=\"ESP_AP\" 
+    -DCONFIG_WIFI_PASSWORD=\"MYPASS\"
+=======
 build_flags =
   -DCONFIG_WIFI_SSID=\"ESP_AP\"
   -DCONFIG_WIFI_PASSWORD=\"MYPASS\"
+>>>>>>> 37d26654a794bd18313ccf1e255576bc9ace1ac2
```
Comment (nopeslide) [inline platformio.ini]: looks like a merge conflict

**Review by nopeslide — CHANGES_REQUESTED** [review]

**Review by ikstream — CHANGES_REQUESTED** [review]
Could you please update the comments, so they match them used in ShilaTu/esp32-core#28

**Comment by jhonnyam** [comment]
This changes where made useless by #32

**Pull request closed without merging**